libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/WaynameFeatureFilter.java

```diff
@@ -0,0 +1,149 @@
+      }
+
+      double smallestUserDistanceToFeature = Double.POSITIVE_INFINITY;
+      for (LineString featureLineString : featureLineStrings) {
+        // 10 meters ahead of current location on the feature LineString
+        Point pointAheadFeature = findPointFromCurrentPoint(currentPoint, TEN_METERS, featureLineString);
+        Timber.d("NAV_DEBUG pointAheadFeature: %s", pointAheadFeature);
+        // 10 meters behind the current location on the feature LineString
+        LineString reversedFeatureLineString = reverseFeatureLineString(featureLineString);
+        Point pointBehindFeature = findPointFromCurrentPoint(currentPoint, TEN_METERS, reversedFeatureLineString);
+        Timber.d("NAV_DEBUG pointBehindFeature: %s", pointBehindFeature);
+        // 10 meters ahead of the current location on the step LineString
+        Point pointAheadUser = findPointFromCurrentPoint(currentPoint, TEN_METERS, currentStepLineString);
+        Timber.d("NAV_DEBUG pointAheadUser: %s", pointAheadUser);
+
+        double userDistanceToAheadFeature = calculateDistance(pointAheadUser, pointAheadFeature);
+        Timber.d("NAV_DEBUG userDistanceToAheadFeature: %s", userDistanceToAheadFeature);
+        double userDistanceToBehindFeature = calculateDistance(pointAheadUser, pointBehindFeature);
+        Timber.d("NAV_DEBUG userDistanceToBehindFeature: %s", userDistanceToBehindFeature);
+        double smallestDistanceToFeature = Math.min(userDistanceToAheadFeature, userDistanceToBehindFeature);
+        Timber.d("NAV_DEBUG smallestDistanceToFeature: %s", smallestDistanceToFeature);
+
+        if (smallestDistanceToFeature < smallestUserDistanceToFeature) {
+          logName(feature);
+          smallestUserDistanceToFeature = smallestDistanceToFeature;
+          filteredFeature = feature;
+        }
+      }
+    }
+    return filteredFeature;
+  }
+
+  private void logNames(List<Feature> queriedFeatures) {
+    List<String> names = new ArrayList<>();
+    for (Feature feature : queriedFeatures) {
+      boolean hasValidNameProperty = feature.hasNonNullValueForProperty("name");
+      if (hasValidNameProperty) {
+        names.add(feature.getStringProperty("name"));
+      }
+    }
+    Timber.d("NAV_DEBUG Queried names: %s", names);
+  }
+
+  private void logName(Feature feature) {
+    boolean hasValidNameProperty = feature.hasNonNullValueForProperty("name");
+    if (hasValidNameProperty) {
+      String name = feature.getStringProperty("name");
+      Timber.d("NAV_DEBUG filteredFeature found: %s", name);
+    }
+  }
+
+  private double calculateDistance(Point pointAheadUser, Point pointAheadFeature) {
+    if (pointAheadUser == null || pointAheadFeature == null) {
+      return Double.POSITIVE_INFINITY;
+    }
+    return TurfMeasurement.distance(pointAheadUser, pointAheadFeature);
+  }
+
+  @Nullable
+  Point findPointFromCurrentPoint(Point currentPoint, double metersFromCurrentPoint, LineString lineString) {
```

Comment:
@1ec5 @vincethecoder I believe I have most of the logic here, minus this method.  iOS is using a chain of extension functions to accomplish what I'm trying to do here in this method.  

This method is called three times starting https://github.com/mapbox/mapbox-navigation-android/pull/1156/files#diff-1e717bd96de82a3e257f8b4d8e92e0a2R69.  Do you see any issues with this logic?  

When it's called it looks like it returns the same values for all three scenarios (feature ahead, feature behind, user ahead):
```
D/WaynameFeatureFilter: NAV_DEBUG pointAheadFeature: Point{type=Point, bbox=null, coordinates=[-77.02962967632158, 38.904698288477306]}
D/WaynameFeatureFilter: NAV_DEBUG pointBehindFeature: Point{type=Point, bbox=null, coordinates=[-77.02962967632158, 38.904698288477306]}
D/WaynameFeatureFilter: NAV_DEBUG pointAheadUser: Point{type=Point, bbox=null, coordinates=[-77.0296298615751, 38.90469829037757]}
```